Realm/RLMRealm_Private.hpp

```diff
@@ -21,6 +21,7 @@
 #import "RLMClassInfo.hpp"
 
 #import <realm/object-store/object_schema.hpp>
+#import <realm/object-store/shared_realm.hpp>
```

Comment:
Can you forward declare a cpp type? I can't seem to @class SharedRealm because it conflicts with the typedef. And if I try to declare it as typedef in the namespace, compiler says all cpp types require type specifier.

Reply:
It should use `std::shared_ptr<Realm>` instead of `SharedRealm`.